client/src/templates/Introduction/components/super-block-accordion.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ReactNode, useMemo } from 'react';
+import { uniq } from 'lodash-es';
+import { useTranslation } from 'react-i18next';
+// TODO: Add this component to freecodecamp/ui and remove this dependency
+import { Disclosure } from '@headlessui/react';
+
+import { ChallengeNode } from '../../../redux/prop-types';
+import { SuperBlocks } from '../../../../../shared/config/curriculum';
+import DropDown from '../../../assets/icons/dropdown';
+import { BlockTypes } from '../../../../../shared/config/blocks';
+import Block from './block';
+
+import './super-block-accordion.css';
+
+interface ChapterProps {
+  dashedName: string;
+  children: ReactNode;
+  isExpanded: boolean;
+}
+
+interface ModuleProps {
+  dashedName: string;
+  children: ReactNode;
+  isExpanded: boolean;
+}
+interface SuperBlockTreeViewProps {
+  challenges: ChallengeNode['challenge'][];
+  superBlock: SuperBlocks;
+  chosenBlock: string;
+}
+
+const Chapter = ({ dashedName, children, isExpanded }: ChapterProps) => {
+  const { t } = useTranslation();
+
+  return (
+    <Disclosure as='li' className='chapter' defaultOpen={isExpanded}>
+      <Disclosure.Button className='chapter-button'>
+        {t(`intro:full-stack-developer.chapters.${dashedName}`)}
+        <DropDown />
+      </Disclosure.Button>
+      <Disclosure.Panel as='ul' className='chapter-panel'>
+        {children}
+      </Disclosure.Panel>
+    </Disclosure>
+  );
+};
+
+const Module = ({ dashedName, children, isExpanded }: ModuleProps) => {
+  const { t } = useTranslation();
+
+  return (
+    <Disclosure as='li' defaultOpen={isExpanded}>
+      <Disclosure.Button className='module-button'>
+        <DropDown />
+        {t(`intro:full-stack-developer.modules.${dashedName}`)}
+      </Disclosure.Button>
+      <Disclosure.Panel as='ul' className='module-panel'>
+        {children}
+      </Disclosure.Panel>
+    </Disclosure>
+  );
+};
+
+export const SuperBlockAccordion = ({
+  challenges,
+  superBlock,
+  chosenBlock
+}: SuperBlockTreeViewProps) => {
+  const content = useMemo(() => {
```

Comment:
`useMemo` can help here, but currently it's not doing a lot. While `challenges` never changes, `chosenBlock` does and each time it does we recalculate a few things we don't need to.

So, rather than putting the whole thing in a memo, we can memoize just the parts that depend on challenges.

I'll push a commit in a bit so you can see what I'm getting at.